bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/coolbar/MandantSelectionContributionItem.java

```diff
@@ -188,6 +174,52 @@ public void widgetSelected(SelectionEvent e) {
 		return toolbar;
 	}
 
+	private void rebuildMenuItems(List<String> workingForIds) {
+		if (menuItems != null && menuItems.length > 0) {
+			for (int i = 0; i < menuItems.length; i++) {
+				if (menuItems[i] != null) {
+					menuItems[i].dispose();
+				}
+			}
+		}
+		if (menu != null && menu.isDisposed()) {
+			menu.dispose();
+		}
+
+		menu = new Menu(fParent);
+		List<MenuItem> menuItemList = new ArrayList<>();
+
+		for (IMandator m : mandants) {
+			if (workingForIds.contains(m.getId())) {
+				MenuItem item = buildMenuItem(m);
+				menuItemList.add(item);
+			}
+		}
+		menuItems = menuItemList.toArray(new MenuItem[0]);
+	}
+
+	private MenuItem buildMenuItem(IMandator m) {
+		MenuItem menuItem = new MenuItem(menu, SWT.RADIO);
+		menuItem.setText(m.getLabel());
+		Image img = getBoxSWTColorImage(UiMandant.getColorForIMandator(m));
+		if (img != null && !img.isDisposed()) {
+			menuItem.setImage(img);
+		}
+		menuItem.setData(m.getId());
+		menuItem.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				ContextServiceHolder.get().setActiveMandator(m);
+			}
+		});
+
+		IMandator activeMandator = ContextServiceHolder.getActiveMandatorOrNull();
+		if (activeMandator != null) {
+			menuItem.setSelection(activeMandator != null && activeMandator.equals(m));
```

Comment:
Die Prüfung activeMandator != null hier bitte entfernen, die ist bereits in der Zeile davor.